vcd/resource_vcd_org_vdc_access_control_test.go

```diff
@@ -0,0 +1,189 @@
+	}
+
+	orgUserPasswordFile := "org_user_pwd.txt"
+	userName1 := strings.ToLower(t.Name())
+	userName2 := strings.ToLower(t.Name()) + "2"
+	accessControlName := "test-access-control"
+
+	var params = StringMap{
+		"Org":               testConfig.VCD.Org,
+		"Vdc":               testConfig.VCD.Vdc,
+		"AccessControlName": accessControlName,
+		"UserName":          userName1,
+		"UserName2":         userName2,
+		"PasswordFile":      orgUserPasswordFile,
+		"RoleName":          govcd.OrgUserRoleOrganizationAdministrator,
+	}
+
+	params["FuncName"] = t.Name() + "step1"
+	configText := templateFill(testAccCheckVcdAccessControlStep1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	params["FuncName"] = t.Name() + "step2"
+	configTex2 := templateFill(testAccCheckVcdAccessControlStep2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckVDCControlAccessDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					assertVdcAccessControlIsSharedWithEverybody(),
+				),
+			},
+			{
+				Config: configTex2,
+				Check: resource.ComposeTestCheckFunc(
+					assertVdcAccessControlIsSharedWithSpecificUser(userName1),
+					assertVdcAccessControlIsSharedWithSpecificUser(userName2),
+				),
+			},
+			{
+				ResourceName:      fmt.Sprintf("vcd_org_vdc_access_control.%s", accessControlName),
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateId:     testConfig.VCD.Org + "." + testConfig.VCD.Vdc,
+			},
+		},
+	})
+}
+
+const testAccCheckVcdAccessControlStep1 = `
+resource "vcd_org_vdc_access_control" "{{.AccessControlName}}" {
+  org                   = "{{.Org}}"
+  vdc                   = "{{.Vdc}}"
+  shared_with_everyone  = true
+  everyone_access_level = "ReadOnly"
+}
+`
+
+const testAccCheckVcdAccessControlStep2 = `
+resource "vcd_org_user" "{{.UserName}}" {
+  org            = "{{.Org}}"
+  name           = "{{.UserName}}"
+  password_file  = "{{.PasswordFile}}"
+  role           = "{{.RoleName}}"
+  take_ownership = true
+}
+
+resource "vcd_org_user" "{{.UserName2}}" {
+  org            = "{{.Org}}"
+  name           = "{{.UserName2}}"
+  password_file  = "{{.PasswordFile}}"
+  role           = "{{.RoleName}}"
+  take_ownership = true
+}
+
+resource "vcd_org_vdc_access_control" "{{.AccessControlName}}" {
+  org                   = "{{.Org}}"
+  vdc                   = "{{.Vdc}}"
+  shared_with_everyone  = false
+
+  shared_with {
+    user_id             = vcd_org_user.{{.UserName}}.id
+    access_level        = "ReadOnly"
+  }
+
+  shared_with {
+    user_id             = vcd_org_user.{{.UserName2}}.id
+    access_level        = "ReadOnly"
+  }
+}
+`
+
```

Comment:
update test missing

Reply:
Test have three steps. First one creates the resource `vcd_org_vdc_access_control` and the second steps updates that resource.